tests/integration/contacts/contact-edit.spec.js

```diff
@@ -0,0 +1,42 @@
+const utils = require('../../utils');
+const personFactory = require('../../factories/cht/contacts/person');
+const place = require('../../factories/cht/contacts/place');
+const {expect} = require('chai');
+
+describe('Editing contacts ', () => {
+  const district = place.generateHierarchy(['district_hospital'])[0];
```

Comment:
This is a convince method to generate a simple hierarchy.  Since you are not using hierarchy it is a bit preferred to use the build function. 

With the rename above. 

`const district = placeFactory.place.build({type:'district_hospital'});`

If you find we end up doing this a lot you could add a couple convince functions to place for the specific types. Then you could do 

`const district = placeFactory.districtHospital();`

`const healthCenter = placeFactory.healthCenter();`

`const area =  placeFactory.area();`

Reply:
Agree - we need to refactor these factories and make them a little bit more versatile. Maybe in another PR .

Reply:
I think this change still needs to be applied. 

I'm in the midst of adding the primary contacts to be generated when calling `generateHierarchy()`. After the change the order might not have 0 index as the `district_hospital`  anymore.

`const district = placeFactory.place.build({type:'district_hospital'});`

Reply:
`placeFactory.generateHierarchy(['district_hospital'])` will generate a place of type `district_hospital` but in array form, which is why there is that index 0.

```
[
  {
    name: 'district hospital0',
    type: 'district_hospital',
    _id: 'e68c720d-4d3e-49b6-8f95-13fd78edb480',
    parent: '',
    is_name_generated: 'true',
    external_id: '',
    notes: '',
    place_id: '35ff68cd-29a0-4e1e-9a83-ac086fd67edf',
    reported_date: 2021-08-03T18:41:23.492Z
  },
  contact: {
    parent: { _id: undefined, parent: undefined },
    _id: '2929cfa9-3316-431f-a9eb-64b394484a42',
    type: 'person',
    name: 'Mary Smith',
    short_name: '',
    date_of_birth_method: 'approx',
    date_of_birth: '2000-02-01',
    ephemeral_dob: {
      dob_calendar: '2000-02-01',
      ephemeral_months: 3,
      ephemeral_years: 2021,
      dob_approx: '2021-03-30',
      dob_raw: '2000-02-01',
      dob_iso: '2000-02-01'
    },
    sex: 'female',
    phone: '',
    phone_alternate: '',
    patient_id: 'test_woman_1',
    reported_date: 2021-08-03T18:41:23.492Z
  }
]
```

  `placeFactory.place().build({type:'district_hospital'})` produces the same thing as `placeFactory.generateHierarchy(['district_hospital'])[0]` (almost): 
```
{
  type: 'district_hospital',
  _id: '301fe443-a7b2-474f-a7bd-fb3fc1d339d7',
  parent: '',
  is_name_generated: 'true',
  name: 'A Place 1',
  external_id: '',
  notes: '',
  place_id: '3aaa841d-8792-423f-8cfc-61f6ee855ab8',
  reported_date: 2021-08-03T18:37:29.912Z,
  contact: {
    parent: { _id: '301fe443-a7b2-474f-a7bd-fb3fc1d339d7', parent: '' },
    _id: '08b69bef-3b8c-4756-aa24-17be0ec128f9',
    type: 'person',
    name: 'Mary Smith',
    short_name: '',
    date_of_birth_method: 'approx',
    date_of_birth: '2000-02-01',
    ephemeral_dob: {
      dob_calendar: '2000-02-01',
      ephemeral_months: 3,
      ephemeral_years: 2021,
      dob_approx: '2021-03-30',
      dob_raw: '2000-02-01',
      dob_iso: '2000-02-01'
    },
    sex: 'female',
    phone: '',
    phone_alternate: '',
    patient_id: 'test_woman_1',
    reported_date: 2021-08-03T18:37:29.912Z
  }
}
```
They both seem to pass, and the change you are suggesting do not alter that, as the index does not refer to the order places are generated. In any case, I went ahead and used  `placeFactory.place().build({type:'district_hospital'})` to get the ticket moving.